crates/lib/src/bootc_composefs/state.rs

```diff
@@ -0,0 +1,60 @@
+use std::{
+    fs::{create_dir_all, remove_dir_all},
+    process::Command,
+};
+
+use anyhow::{Context, Result};
+use camino::Utf8PathBuf;
+use fn_error_context::context;
+
+use rustix::{
+    fs::{open, Mode, OFlags, CWD},
+    mount::{unmount, UnmountFlags},
+};
+
+/// Mounts an EROFS image and copies the pristine /etc to the deployment's /etc
+#[context("Copying etc")]
+pub(crate) fn copy_etc_to_state(
+    sysroot_path: &Utf8PathBuf,
+    erofs_id: &String,
+    state_path: &Utf8PathBuf,
+) -> Result<()> {
+    let sysroot_fd = open(
+        sysroot_path.as_std_path(),
+        OFlags::PATH | OFlags::DIRECTORY | OFlags::CLOEXEC,
+        Mode::empty(),
+    )
+    .context("Opening sysroot")?;
+
+    let composefs_fd = bootc_initramfs_setup::mount_composefs_image(&sysroot_fd, &erofs_id, false)?;
+
+    let uuid = uuid::Uuid::new_v4();
+    let dir = format!("/var/tmp/{uuid}");
+
+    create_dir_all(&dir).context("Creating temp directory")?;
+
+    bootc_initramfs_setup::mount_at_wrapper(composefs_fd, CWD, &dir)?;
```

Comment:
Ah yeah, we have mount helper code in lib/...and in composefs-boot, probably again need to factor out a helper crate